executor/window.go

```diff
@@ -125,11 +149,7 @@ func (e *WindowExec) consumeGroupRows(groupRows []chunk.Row) (err error) {
 	return nil
 }
 
-func (e *WindowExec) fetchChildIfNecessary(ctx context.Context) (EOF bool, err error) {
-	if e.inputIter != nil && e.inputIter.Current() != e.inputIter.End() {
-		return false, nil
-	}
-
+func (e *WindowExec) fetchChild(ctx context.Context) (EOF bool, err error) {
```

Comment:
How about changing the output arguments to `(result *Chunk, EOF bool, err error)` then we can remove the `childResult` field in `WindowExec` which makes this executor simpler.

Reply:
Rest LGTM